libpod/runtime_ctr.go

```diff
@@ -645,10 +645,14 @@ func (r *Runtime) removeContainer(ctx context.Context, c *Container, force, remo
 		}
 	} else {
 		if err := r.state.RemoveContainer(c); err != nil {
-			if cleanupErr == nil {
-				cleanupErr = err
+			if err == define.ErrNoSuchCtr {
```

Comment:
I think this is still worth an error. If we got this far, we did it *holding the lock* - there is 0 chance the container was removed from under us. If this happens something very strange and very bad is going on - someone is messing around in the DB without obeying our locks.

Reply:
Ok some of this was just guessing.  I will remove this check.

Reply:
Actually @mheon @edsantiago reported in the original issue that he was getting.
WARN[0000] Container 6b4d7060b9466bd7c0e806758b3c2366227ff415784541882ea324c0bfb1b9e2 does not exist: container 6b4d7060b9466bd7c0e806758b3c2366227ff415784541882ea324c0bfb1b9e2 does not exist in database: no such container

Which looks like this is happening.

Reply:
Then we have a serious bug, because that should never happen.

Reply:
Ok I reverted, let's see if this error is reproducible, with the other fixes.

Reply:
I've pulled your latest change (6ca721cccd98bf9936b33d690b28a52cd4213a18), rebuilt, and am running tests. No failures in the last two minutes, which is good. I need to be AFK periodically this morning but will check in as time allows.

Reply:
Still running with no errors (20+ minutes). I call that good. Gotta head out now, will check back in in 90m or so.

Reply:
Still no failures. I call this good.